src/OAuth2Form.tsx

```diff
@@ -1,384 +1,447 @@
-                }
+  const getUserProfile = () => {
+    if (accessToken && !profile && !error) {
+      if (!isGettingUserProfile) {
+        setIsGettingUserProfile(true);
+      }
+    }
+  };
+
+  useEffect(() => {
+    let isSubscribed = true;
+    if (accessToken && isGettingUserProfile) {
+      getHasAlreadyConsented(accessToken);
+      SynapseClient.getUserProfile(accessToken)
+        .then((profile: UserProfile) => {
+          if (isSubscribed) {
+            if (profile.profilePicureFileHandleId) {
+              profile.clientPreSignedURL = `https://www.synapse.org/Portal/filehandleassociation?associatedObjectId=${profile.ownerId}&associatedObjectType=UserProfileAttachment&fileHandleId=${profile.profilePicureFileHandleId}`;
             }
-        }
+            setProfile(profile);
+          }
+        })
+        .catch((_err) => {
+          onError(_err);
+        });
     }
-
-    useEffect(() => {
-        let isSubscribed = true
-        if (isGettingUserProfile) {
-            const { accessToken: newToken } = props.context!
-            getHasAlreadyConsented(newToken!)
-            setToken(newToken)
-            SynapseClient.getUserProfile(newToken).then((profile: UserProfile) => {
-                if (isSubscribed) {
-                    if (profile.profilePicureFileHandleId) {
-                        profile.clientPreSignedURL = `https://www.synapse.org/Portal/filehandleassociation?associatedObjectId=${profile.ownerId}&associatedObjectType=UserProfileAttachment&fileHandleId=${profile.profilePicureFileHandleId}`
-                    }
-                    setProfile(profile)
-                }
-            }).catch((_err) => {
-                onError(_err)
-            })
-        }
-        return () => {
-            isSubscribed = false
-        }
-    }, [isGettingUserProfile])
-
-    /**
-       * Returns scopes UI.  All the links formatted accordingly
-       */
-    const Scopes = () => {
-        return <>
-            {oidcRequestDescription && <ul>
-                {
-                    oidcRequestDescription.scope.map(
-                        (scope, index) => {
-                            return (
-                                <li key={index}>
-                                    {scope}
-                                </li>
-                            )
-                        })
-                }
-            </ul>}
-        </>
+    return () => {
+      isSubscribed = false;
+    };
+  }, [isGettingUserProfile]);
+
+  /**
+   * Returns scopes UI.  All the links formatted accordingly
+   */
+  const Scopes = () => {
+    return (
+      <>
+        {oidcRequestDescription && (
+          <ul>
+            {oidcRequestDescription.scope.map((scope, index) => {
+              return <li key={index}>{scope}</li>;
+            })}
+          </ul>
+        )}
+      </>
+    );
+  };
+
+  //init
+  useEffect(() => {
+    getUserProfile();
+    getOauthClientInfo();
+  }, [accessToken]);
```

Comment:
Since the token passed via context is now the source of truth, we can just use it to drive this effect.